src/coreclr/jit/emitriscv64.cpp

```diff
@@ -13,6 +13,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "jitpch.h"
 #ifdef _MSC_VER
 #pragma hdrstop
+#pragma warning(disable : 4244 4267)
```

Comment:
There are a number of "possible loss of data" warnings related to the helpers like `NBitMask` that required adding a bunch of casts in many places in this file. I went with just disabling these warnings since I did not want to uglify all of that code. Perhaps there is a prettier way to fix these warnings.

Reply:
Disabling this warning is SDL violation: https://github.com/dotnet/runtime/blob/ac07ea6583cfa7e6896c2fe97b595f2a395ed3ea/eng/native/configurecompiler.cmake#L847

It is only ok as long as this code is not getting compiled in official builds.

Reply:
@jakobbotsch Thank you.
@Bajtazar Could you fix the warnings in `emitriscv64.cpp`? The most of warnings are casts related to helpers which you have refactoring. I think you can fix in prettier way. :)

Reply:
Hi, I prepared a [patch](https://[patch-diff.githubusercontent.com/raw/Bajtazar/runtime/pull/1.patch](https://patch-diff.githubusercontent.com/raw/Bajtazar/runtime/pull/1.patch)) that fixes warnings in the `emitriscv64.cpp` file

Reply:
@Bajtazar I think the patch can be merged separately. Could you make another PR for your patch?

Reply:
Thank you!

> @Bajtazar I think the patch can be merged separately. Could you make another PR for your patch?

You can open a separate PR or I can include it in this change. Up to you.

Reply:
@clamp03 @jakobbotsch I've opened a separate PR with fixes [here](#100270)